apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaProducerManager.java

```diff
@@ -153,6 +156,18 @@ private void notifyListeners(KafkaConnectionStatus status) {
         }
     }
 
+    private Map<String, String> decrypt(Map<String, String> config) {
+        try {
+            Class<?> decryptClazz = Class.forName(Kafka.DECRYPT_CLASS);
+            Method decryptMethod = decryptClazz.getMethod(Kafka.DECRYPT_METHOD, Map.class);
+            return (Map<String, String>) decryptMethod.invoke(decryptClazz.newInstance(), config);
```

Comment:
I don't think reflection for the whole process is a good idea. You should add an interface in the reporter project,

```java
public interface KafkaConfigExtension {
    Map<String, String> decode(Map<String, String> config);
}
```

Then this interface would be released, and user could simply depend on and package. Reflection should be something that has to be that way.

Notice, your saying `decrypt` actually is just `decode`. Because decrypting usually means you have in-built algorithms.